src/coresyf_image_crop/tests/test_image_crop.py

```diff
@@ -37,3 +36,19 @@ def test_get_shapefile_crs(self):
         epsg_code = get_shapefile_crs(self.data_source)
         self.assertEqual(epsg_code, self.epsg_code)
 
+    def test_apply_buffer_to_polygon(self):
+        buffer_crs = get_raster_crs(gdal.Open(self.test_image_4236))
+        polygon_extent = get_shapefile_polygon_extent(self.data_source)
+        buffer_4326 = 0.022207
+        buffer_3763 = 2000
+        polygon_crs = get_shapefile_crs(self.data_source)
+
+        polygon_with_buffer = apply_buffer_to_polygon(buffer_4326, polygon_extent,
+                                                      buffer_crs, polygon_crs)
+
+        expected_polygon = ogr.CreateGeometryFromWkt(self.extent_polygon_wkt)
```

Comment:
I didn't saw the code used to define this `self.extent_polygon_wkt` but I suppose we can apply exactly the same approach to create the input polygon. We do not have to obtain it from an actual raster. Remember that what we are testing here is just that we are able to add a buffer to a polygon, not that such approach works when we apply it to our main program.

Reply:
Indeed a simple way to test is: we create a polygon, then we make a copy and apply a negative buffer (if such is possible); then we add the absolute value of the previous buffer. If this polygon is equal to the original polygon, we are done!